pkgs/development/python-modules/hdbscan/default.nix

```
@@ -2,40 +2,30 @@
 , buildPythonPackage
 , fetchpatch
 , cython
-, numpy
+, oldest-supported-numpy
 , pytestCheckHook
 , scipy
 , scikit-learn
 , fetchPypi
 , joblib
+, setuptools
 , six
 , pythonRelaxDepsHook
 }:
 
 buildPythonPackage rec {
   pname = "hdbscan";
   version = "0.8.33";
+  pyproject = true;
 
   src = fetchPypi {
     inherit pname version;
     hash = "sha256-V/q8Xw5F9I0kB7NccxGSq8iWN2QR/n5LuDb/oD04+Q0=";
   };
-  patches = [
-    # should be included in next release
-    (fetchpatch {
-      name = "joblib-1.2.0-compat.patch";
-      url = "https://github.com/scikit-learn-contrib/hdbscan/commit/d829c639923f6866e1917e46ddbde45b513913f3.patch";
-      excludes = [
-        "docs/basic_hdbscan.rst"
-        "docs/how_hdbscan_works.rst"
-      ];
-      hash = "sha256-t0D4OsHEcMwmBZM8Mk1N0uAKi6ra+TOzEks9/efsvWI=";
-    })
-  ];
 
   pythonRemoveDeps = [ "cython" ];
-  nativeBuildInputs = [ pythonRelaxDepsHook cython ];
-  propagatedBuildInputs = [ numpy scipy scikit-learn joblib six ];
+  nativeBuildInputs = [ pythonRelaxDepsHook cython setuptools ];
+  propagatedBuildInputs = [ oldest-supported-numpy scipy scikit-learn joblib six ];
```

```suggestion
  nativeBuildInputs = [ pythonRelaxDepsHook cython setuptools oldest-supported-numpy ];
  propagatedBuildInputs = [ numpy scipy scikit-learn joblib six ];
```

missed this. oldest-supported-numpy is a build input (i think it just resolves to numpy).